model/VocabularyConfig.php

```diff
@@ -31,41 +31,97 @@ class VocabularyConfig extends BaseConfig
     public function __construct($resource, $globalPlugins=array())
     {
         $this->resource = $resource;
-        $plugins = $this->resource->allLiterals('skosmos:usePlugin');
+        $this->globalPlugins = $globalPlugins;
+        $this->setParametrizedPlugins();
+        $pluginArray = $this->getPluginArray();
+        $this->pluginRegister = new PluginRegister($pluginArray);
+    }
+
+    /**
+     * Get an ordered array of plugin names with order configured in skosmos:vocabularyPlugins
+     * @return array|null of plugin names
+     */
+    public function getPluginArray() {
+
         $pluginArray = array();
+
+        $vocabularyPlugins = $this->resource->getResource('skosmos:vocabularyPlugins');
+        if ($vocabularyPlugins) {
+            // $vocabularyPlugins has all resources
+            foreach ($vocabularyPlugins as $plugin) {
+                if ($plugin instanceof EasyRdf\Literal) {
+                    $pluginName = $plugin->getValue();
+                    array_push($pluginArray, $pluginName);
+                }
+                else {
+                    array_push($pluginArray, $plugin->getLiteral('skosmos:usePlugin')->getValue());
+                }
+            }
+        }
+        $pluginArray = array_merge($pluginArray, $this->globalPlugins);
+
+        $plugins = $this->resource->allLiterals('skosmos:usePlugin');
         if ($plugins) {
             foreach ($plugins as $pluginlit) {
-                $pluginArray[] = $pluginlit->getValue();
+                $pluginName = $pluginlit->getValue();
+                array_push($pluginArray, $pluginName);
             }
         }
-        $this->plugins = new PluginRegister(array_merge($globalPlugins, $pluginArray));
-        // Get parameterized plugins defined as resources and their respective parameters
-        $pluginResources = $this->resource->allResources('skosmos:useParamPlugin');
+
+        $pluginArray = array_values(array_unique(array_merge($pluginArray, array_keys($this->pluginParameters))));
+
+        return $pluginArray;
+
+    }
+
+    /**
+     * Sets array of parameterized plugins
+     * @param Easyrdf\Resource $pluginResource
+     */
+    private function setParametrizedPlugins() {
```

Comment:
one could add `@return void` and `:void` here

Reply:
Also `paramaterizedPlugins` instead of `parametrizedPlugins` (need to be fixed elsewhere too)